sdk/sync/write.go

```diff
@@ -76,25 +76,31 @@ func (w *Writer) keepAliveWorker(ctx context.Context) {
 	for {
 		select {
 		case <-time.After(KeepAlivePeriod):
-			w.keepAlive(ctx)
+			if err := w.keepAlive(ctx); err != nil {
+				if ctx.Err() != nil {
```

Comment:
Could possibly use:

```go
if errors.Is(err, context.Canceled)
```

Reply:
So... I don't trust errors. I'm seeing an "invalid argument" error which could, maybe, be caused by:

```go
go func() {
  <-ctx.Done()
  myConn.Close()
}()
return myConn.Read()
```

(depending on the connection type, etc.)